editor/store/effects.js

```diff
@@ -175,6 +176,14 @@ export default {
 			) );
 		}
 
+		// The server can return a sanitised version of the slug,
+		// in which case we need to update our local copy.
+		if ( get( edits, 'slug' ) !== post.slug ) {
+			dispatch( {
+				type: 'POSTNAME_SANITIZED',
+			} );
```

Comment:
If I understand correctly:

- `RESET_POST` is meant to represent "here's the server-persisted copy of the post". Introducing `edits` muddies this meaning.
- Instead, we could do similar unsetting of "now persisted" edits as part of an `UPDATE_POST` handler in the `state.editor.edits` reducer, since this includes and is meaningfully tied to the `edits` fields.

Maybe something like:

```js
case 'UPDATE_POST':
case 'RESET_POST':
	const getCanonicalValue = action.type === 'UPDATE_POST' ?
		( key ) => action.edits[ key ] :
		( key ) => getPostRawValue( action.post[ key ] );

	return reduce( state, ( result, value, key ) => {
		if ( value !== getCanonicalValue( key ) ) {
			return result;
		}

		if ( state === result ) {
			result = { ...state };
		}

		delete result[ key ];
		return result;
	}, state );
```

Reply:
Nice, that looks much better. 🙂